core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -297,6 +283,38 @@ protected void validateAddedDataFiles(TableMetadata base, Long startingSnapshotI
     }
   }
 
+  /**
+   * Validates that no new delete files have been added to the table since a starting snapshot.
```

Comment:
I think this is no new delete files of _the specified data files_ have been added to the table.

Reply:
I updated the wording here to mention that it tests the data files.